src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFsDriver.Linux.cs

```diff
@@ -24,12 +24,18 @@ public class SysFsDriver : UnixDriver
         private readonly CancellationTokenSource _eventThreadCancellationTokenSource;
         private readonly List<int> _exportedPins = new List<int>();
         private readonly Dictionary<int, UnixDriverDevicePin> _devicePins = new Dictionary<int, UnixDriverDevicePin>();
-        private readonly int _pollingTimeoutInMilliseconds = Convert.ToInt32(TimeSpan.FromMilliseconds(1).TotalMilliseconds);
-        private static readonly int s_pinOffset = ReadOffset();
+        private static readonly int s_pinOffset;
+        private int _pollingTimeoutInMilliseconds = Convert.ToInt32(TimeSpan.FromMilliseconds(1).TotalMilliseconds);
+        private int _statusUpdateSleepTime = Convert.ToInt32(TimeSpan.FromMilliseconds(1).TotalMilliseconds);
         private int _pollFileDescriptor = -1;
         private Thread _eventDetectionThread;
         private int _pinsToDetectEventsCount;
 
+        static SysFsDriver()
+        {
+            s_pinOffset = ReadOffset();
```

Comment:
Any reason why we chose to have a static constructor here instead? In general we try to avoid them when possible since they can have unintended effects like by changing completely the way that types are constructed. If there is no specific reason why we moved it to be like this, I would keep it as we had it with just doing an inline definition above.

Reply:
Not 100% sure, but the previous line `private static readonly int s_pinOffset = ReadOffset();` supposedly creates the exact same code. I tend to write explicit static ctors when the operation is lengthy.

Reply:
You are right that the effect it has is very similar, except that it is much easier to understand and control the execution time of the previous way the code was, than static constructors. Those run at an incontrolable time and may have unintended effects if you had any other static fields that you were auto setting, or if you had subclasses that did. Also, it makes the code a bit less readable, so if there is no specific reason why we really want static constructors, I would suggest to avoid them.